File: php/features.php
<?php
declare(strict_types=1);

// QrCodeGenerator SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class QrCodeGeneratorFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new QrCodeGeneratorBaseFeature();
            case "test":
                return new QrCodeGeneratorTestFeature();
            default:
                return new QrCodeGeneratorBaseFeature();
        }
    }
}

File: py/features.py
# QrCodeGenerator SDK feature factory

from feature.base_feature import QrCodeGeneratorBaseFeature
from feature.test_feature import QrCodeGeneratorTestFeature


def _make_feature(name):
    features = {
        "base": lambda: QrCodeGeneratorBaseFeature(),
        "test": lambda: QrCodeGeneratorTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
